rules/jsxExpressionSpacingRule.ts

```diff
@@ -0,0 +1,68 @@
+﻿import * as Lint from 'tslint/lib/lint';
+import * as ts from 'typescript';
+
+export class Rule extends Lint.Rules.AbstractRule {
+	public apply(sourceFile: ts.SourceFile) {
+		return this.applyWithWalker(new JsxExpressionSpacingWalker(sourceFile, this.getOptions()));
+	}
+}
+
+class JsxExpressionSpacingWalker extends Lint.RuleWalker {
+	public walkChildren(node: ts.Node) {
+		if (node.kind === ts.SyntaxKind.JsxExpression) {
+			this.validateNode(node as ts.JsxExpression);
+		}
+		super.walkChildren(node);
+	}
+
+	private validateNode(node: ts.JsxExpression) {
+		let name: string;
+		if (node.parent.kind === ts.SyntaxKind.JsxAttribute) {
+			name = `'${(node.parent as ts.JsxAttribute).name.getText()}'`;
+		} else {
+			name = (node as ts.JsxExpression).getText();
+		}
+
+		const value = node.getChildAt(1);
+		const closingBrace = node.getChildAt(2);
+
+		if (!value || !this.isClosingBrace(closingBrace)) {
+			return;
+		}
+
+		if (
+			!this.isPrecededByValidWhitespace(value)
+		) {
+			this.addFailure(
+				this.createFailure(
+					node.getStart(),
+					node.getWidth(),
+					`jsx expression ${name} should have one space after opening '{'`
+				)
+			);
+		}
+
+		if (
+			!this.isPrecededByValidWhitespace(closingBrace)
+		) {
+			this.addFailure(
+				this.createFailure(
+					node.getStart(),
+					node.getWidth(),
+					`jsx expression ${name} should have one space before closing '}'`
+				)
+			);
+		}
+	}
+
+	private isPrecededByValidWhitespace(node: ts.Node) {
+		return (
+			node.getFullStart() === node.getStart() - 1 ||
+			node.getFullText().match(/^[\r\n]+/)
```

Comment:
Let's go ahead and switch this to just look at `\n`

We also might need to only look before the node; I'm not sure how TS's tokenizer overlaps whitespace. i.e. this might pass:
```jsx
<div value={ asdf}
   />
```